From 0c50f59bf966b3acbb1d1763be3461b6601fb409 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Mon, 22 Sep 2025 11:06:14 +0200
Subject: [PATCH 178/553] Mark QStringList as security-significant (=default)

While QStringList contains a lot of specialized algorithms on not only
the list itself, but also the QString members, a reasonable user will
expect them to be convenience calling the resp. functions for each
member.

As such, the security implications inherent to QString manipulation
are considered firewalled off by QString (which is one of the reasons
to mark QString as security-critical: so that higher-level code can
assume it's "doing it's job"), and what remains is a QList, which is
(or ought soon to be) marked as significant-only.

All of the above: ditto QByteArrayList.

Task-number: QTBUG-135195
Pick-to: 6.8
Change-Id: I6c791ba59d1727d413f553d6498804431451de4c
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 3ba8e2fbc483f8f5564bb30ee7437a9150e193b5)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qbytearraylist.cpp | 1 +
 src/corelib/text/qbytearraylist.h   | 1 +
 src/corelib/text/qstringlist.cpp    | 1 +
 src/corelib/text/qstringlist.h      | 1 +
 4 files changed, 4 insertions(+)

diff --git a/src/corelib/text/qbytearraylist.cpp b/src/corelib/text/qbytearraylist.cpp
index 81bb54dfe4d..140f953722c 100644
--- a/src/corelib/text/qbytearraylist.cpp
+++ b/src/corelib/text/qbytearraylist.cpp
@@ -1,6 +1,7 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // Copyright (C) 2014 by Southwest Research Institute (R)
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qbytearraylist.h>
 
diff --git a/src/corelib/text/qbytearraylist.h b/src/corelib/text/qbytearraylist.h
index cfae7f5bdff..1ffa4afe104 100644
--- a/src/corelib/text/qbytearraylist.h
+++ b/src/corelib/text/qbytearraylist.h
@@ -2,6 +2,7 @@
 // Copyright (C) 2016 Intel Corporation.
 // Copyright (C) 2014 by Southwest Research Institute (R)
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <QtCore/qlist.h>
 
diff --git a/src/corelib/text/qstringlist.cpp b/src/corelib/text/qstringlist.cpp
index e6f85849f27..dddc119fd2a 100644
--- a/src/corelib/text/qstringlist.cpp
+++ b/src/corelib/text/qstringlist.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qstringlist.h>
 #if QT_CONFIG(regularexpression)
diff --git a/src/corelib/text/qstringlist.h b/src/corelib/text/qstringlist.h
index a8135f2f1ba..02515c8b1c2 100644
--- a/src/corelib/text/qstringlist.h
+++ b/src/corelib/text/qstringlist.h
@@ -1,6 +1,7 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // Copyright (C) 2016 Intel Corporation.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <QtCore/qlist.h>
 
-- 
2.51.2

